api/include/opentelemetry/nostd/string_view.h

```diff
@@ -172,3 +177,18 @@ inline std::ostream &operator<<(std::ostream &os, string_view s)
 }
 }  // namespace nostd
 OPENTELEMETRY_END_NAMESPACE
+
+namespace std
+{
+template <>
+struct hash<OPENTELEMETRY_NAMESPACE::nostd::string_view>
```

Comment:
Is this necessary for VS2015?

Reply:
No, sorry, this is a template that allows to use `nostd::string_view` as key in the `std::map<nostd::string_view, ...>` .. It's adding the missing functionality that otherwise exists for `std::string_view`

Reply:
That's what I meant under the `couple helper routines` umbrella in the PR description 😄

Reply:
Practical reason why I'm adding this: I'm working on a swap between `nostd::` vs. `std::` vs. `absl::` ... Abseil will actually be required for Visual Studio 2015 (that's the only way how I can make backport of `std::variant` work). Some changes like this one, and adding a few headers here and there - are aligning overall code to potentially work with any replacement / substitute for `nostd::` . Once this PR is merged, I'll follow-up to refresh my other PR with support for:
- `nostd::` - default for anything vs2017+ and other OS
- `std::` with msgsl - option for any compiler that supports C++17 and above
- `absl::` - exotic flavor, currently only required for vs2015 (at least for the proper variant backport).

We can also potentially discuss that maybe we should switch from our `nostd::variant` to `absl::variant`, as Abseil's variant is at least definitely working on older compilers where ours is failing.

Reply:
I currently have all tests passing for `nostd::`, `std::` + `absl::` for Vs2015 in my branch here:
https://github.com/maxgolov/opentelemetry-cpp
I'm doing a matrix build with OpenTelemetry-provided containers vs. Standard Library Containers vs. Abseil (for vs2015 only).